pygmt/src/rose.py

```diff
@@ -0,0 +1,235 @@
+          direction and other statistics, use
+          ``vectors = '+wmode_file'``. The eight items saved to
+          a single record are: *mean_az*, *mean_r*, *mean_resultant*,
+          *max_r*, *scaled_mean_r*, *length_sum*, *n*, *sign@alpha*,
+          where the last term is 0 or 1 depending on whether the mean
+          resultant is significant at the level of confidence set via
+          **conf_level**.
+
+    vector_params : str
+        Used with **vectors** to modify vector parameters. For
+        vector heads, append vector head size [Default is 0, i.e., a
+        line]. See :gmt-docs:`rose.html#vector-attributes` for
+        specifying additional attributes. If **vectors** is not
+        given and the current plot mode is to draw a windrose diagram
+        then using **vector_params** will add vector heads to all
+        individual directions using the supplied attributes.
+
+    alpha : float or str
+        Sets the confidence level used to determine if the mean
+        resultant is significant (i.e., Lord Rayleigh test for
+        uniformity) [``alpha = 0.05``]. Note: The critical
+        values are approximated [Berens, 2009] and requires at
+        least 10 points; the critical resultants are accurate to at
+        least 3 significant digits. For smaller data sets you
+        should consult exact statistical tables.
+
+        Berens, P., 2009, CircStat: A MATLAB Toolbox for Circular
+        Statistics, *J. Stat. Software*, 31(10), 1-21,
+        https://doi.org/10.18637/jss.v031.i10.
+
+    inquire : bool
+        Inquire. Computes statistics needed to specify a useful
+        **frame**. No plot is generated. The following statistics
+        are written to stdout: *n*, *mean az*, *mean r*, *mean
+        resultant length*, *max bin sum*, *scaled mean*, and
+        *linear length sum*.
+
+    {U}
+    {V}
+    {XY}
+    {p}
+    {t}
+    """
+
+    kwargs = self._preprocess(**kwargs)
+
+    kind = data_kind(data, length, azimuth)
+
+    with Session() as lib:
+        # Choose how data will be passed in to the module
+        if kind == "file":
+            file_context = dummy_context(data)
+        elif kind == "matrix":
+            file_context = lib.virtualfile_from_matrix(data)
+        elif kind == "vectors":
+            file_context = lib.virtualfile_from_vectors(
+                np.atleast_1d(length), np.atleast_1d(azimuth)
+            )
+
+        with file_context as fname:
+            arg_str = " ".join([fname, build_arg_string(kwargs)])
+
+            if "I" not in kwargs:
+                lib.call_module("rose", arg_str)
+            # if inquire only, give back statistics about input data
+            else:
+                with GMTTempFile() as outfile:
+                    arg_str = " ".join([arg_str, " ->" + outfile.name])
+
+                    lib.call_module("rose", arg_str)
+                    result = outfile.read().strip()
+                    print(result)
```

Comment:
No, it's required if ``-I`` is set to give back the information about the input data @weiji14 . I tested it both in an interactive session from command line and a Jupyter notebook. Otherwise no output is given back. Other suggestions how to deal with that?

Reply:
I think it should return something, not printing.

Reply:
That was what I originally did, however, I receive the following warning when running ``make lint`` after putting ``return(result)`` instead of ``print(result)``: 

``R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)``. 

Since there should be no output for calls without ``-I`` I'm not sure how to handle that.